modules/caddytls/connpolicy.go

```diff
@@ -280,6 +293,30 @@ func (p *ConnectionPolicy) buildStandardTLSConfig(ctx caddy.Context) error {
 		}
 	}
 
+	if p.InsecureSecretsLog != "" {
+		filename, err := caddy.NewReplacer().ReplaceOrErr(p.InsecureSecretsLog, true, true)
```

Comment:
This enables setting the filename from an env var. But it has to be done explicitly, as this definitely isn't a "feature" we want to enable implicitly and obscurely from the environment!